tinylog-impl/src/main/java/org/tinylog/writers/JsonWriter.java

```diff
@@ -0,0 +1,268 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package org.tinylog.writers;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.MappedByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.charset.Charset;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+
+import org.tinylog.Level;
+import org.tinylog.core.LogEntry;
+import org.tinylog.core.LogEntryValue;
+import org.tinylog.pattern.FormatPatternParser;
+import org.tinylog.pattern.Token;
+import org.tinylog.provider.InternalLogger;
+import org.tinylog.writers.raw.BufferedWriterDecorator;
+import org.tinylog.writers.raw.ByteArrayWriter;
+import org.tinylog.writers.raw.OutputStreamWriter;
+import org.tinylog.writers.raw.SynchronizedWriterDecorator;
+
+/**
+ * Writer for outputting log entries to a log file in JSON format. Already
+ * existing files can be continued.
+ */
+public final class JsonWriter implements Writer {
+	private static final String JSON_OBJECT = "{%n%s%n}";
+	private static final String MESSAGE_PATTERN = "\"message\": \"{message}\"";
+	private static final String TIMESTAMP_PATTERN = "\"timestamp\": \"{date}\"";
+	private static final String METHOD_PATTERN = "\"method\": \"{method}()\"";
+	private static final String LEVEL_PATTERN = "\"level\": \"{level}\"";
+	private static final String CLASS_PATTERN = "\"class\": \"{class}\"";
+	private static final String THREAD_PATTERN = "\"thread\": \"{thread}\"";
+	private static final String NEW_LINE = System.getProperty("line.separator");
+	private static final int BUFFER_SIZE = 1024;
+
+	private Charset charset;
+	private ByteArrayWriter writer;
+	private FileChannel fileChannel;
+	private FileChannel inputChannel;
+
+	private StringBuilder builder;
+	private final Token messageToken;
+	private final Token timestampToken;
+	private final Token methodToken;
+	private final Token levelToken;
+	private final Token classToken;
+	private final Token threadToken;
+
+	private byte[] commaByte;
+	private byte[] bracketOpenByte;
+	private byte[] bracketCloseByte;
+
+	/**
+	 * @throws IOException              File not found or couldn't access file
+	 * @throws IllegalArgumentException Log file is not defined in configuration
+	 */
+	public JsonWriter() throws IOException {
+		this(Collections.<String, String>emptyMap());
+	}
+
+	/**
+	 * @param properties Configuration for writer
+	 * @throws IOException              File not found or couldn't access file
+	 * @throws IllegalArgumentException Log file is not defined in configuration
+	 */
+	public JsonWriter(final Map<String, String> properties) throws IOException {
+
```

Comment:
The redundant empty line can be removed (minor code style violation).

Reply:
Fixed in https://github.com/tinylog-org/tinylog/pull/195/commits/5322e7bbaaa6040dfad72f601cac39c19db08998